### PR TITLE
feat: project setup

### PR DESCRIPTION
**Pull request by pedrohenrique-ql**
### Features

- Added `tailwind`.
- Added `prettier`, `eslint`, `lint-staged`, `husky` and `commit-lint` for style checking
- `Http Client` setup.

### CODE REVIEW

File: .lintstagedrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const ignoredFilenamesWithoutExtension = ['.lintstagedrc', 'postcss.config', 'tailwind.config', 'next.config'];
+
+module.exports = {
+  '*': ['pnpm style:format'],
+  [`**/!(${ignoredFilenamesWithoutExtension.join('|')})*.(j|t)s(x|)`]: ['pnpm lint'],
+  '*.prisma': ['pnpm prisma format'],
```
Comment (diego-aquino) [inline .lintstagedrc.js:6]: Is prisma necessary here?

```suggestion
```

File: .lintstagedrc.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const ignoredFilenamesWithoutExtension = ['.lintstagedrc', 'postcss.config', 'tailwind.config', 'next.config'];
```
Comment (diego-aquino) [inline .lintstagedrc.js:1]:
```suggestion
const ignoredFilenamesWithoutExtension = ['.lintstagedrc', 'postcss.config', 'tailwind.config'];
```

File: src/app/app.component.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { HomeComponent } from './components/home/home.component';
+
+@Component({
+  selector: 'app-root',
+  standalone: true,
+  imports: [RouterOutlet, HomeComponent],
+  templateUrl: './app.component.html',
+  styleUrl: './app.component.css',
+})
+export class AppComponent {
```
Comment (diego-aquino) [inline src/app/app.component.ts:12]: Is it an Angular standard to always suffix component names with `Component`?

File: .lintstagedrc.js (hunk 1 above)
Comment (diego-aquino) [inline .lintstagedrc.js]: Is prisma necessary here?

```suggestion
```

File: .lintstagedrc.js (hunk 2 above)
Comment (diego-aquino) [inline .lintstagedrc.js]:
```suggestion
const ignoredFilenamesWithoutExtension = ['.lintstagedrc', 'postcss.config', 'tailwind.config'];
```

File: eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          prefix: 'app',
+          style: 'camelCase',
+        },
+      ],
+      '@angular-eslint/component-selector': [
+        'error',
+        {
+          type: 'element',
+          prefix: 'app',
+          style: 'kebab-case',
```
Comment (diego-aquino) [inline eslint.config.js:32]: Does this says that all component should start with `app-`? Is it an Angular pattern?

File: src/app/services/abuse-ip.service.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  providedIn: 'root',
+})
+export class AbuseIpService {
+  private abuseIpUrl = environment.abuseIpUrl;
+  private abuseIpApiKey = environment.abuseIpApiKey;
+
+  constructor(private httpClient: HttpClient) {}
+
+  public check(ipAddress: string) {
+    return this.httpClient.get(`${this.abuseIpUrl}/check/${ipAddress}?maxAgeInDays=90`, {
```
Comment (diego-aquino) [inline src/app/services/abuse-ip.service.ts]: Does Angular's HTTP client support a base URL? Just out of curiosity.

File: tailwind.config.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ['./src/**/*.{html,ts}'],
```
Comment (diego-aquino) [inline tailwind.config.js:3]: Thanks!

File: src/environments/environment.development.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export const environment = {
+  production: false,
+  abuseIpUrl: 'https://api.abuseipdb.com/api/v2',
+  abuseIpApiKey: '',
```
Comment (diego-aquino) [inline src/environments/environment.development.ts:4]: All envs are declared in a typescript file? Do you know any way to declare the key privately without commiting it?

Also, this is a frontend application. Does Angular have any way to not expose the key? Just asking out of curiosity. This project will be just an example.

**Review by diego-aquino — CHANGES_REQUESTED** [review]
Good job! I left just a few questions.

File: src/app/app.component.ts (hunk 3 above)
Comment (pedrohenrique-ql) [reply src/app/app.component.ts:12]: Yes

File: .lintstagedrc.js (hunk 1 above)
Comment (pedrohenrique-ql) [reply .lintstagedrc.js]: No, I will remove :joy:

File: eslint.config.js (hunk 4 above)
Comment (pedrohenrique-ql) [reply eslint.config.js:32]: From the content I saw, yes

File: src/app/services/abuse-ip.service.ts (hunk 5 above)
Comment (pedrohenrique-ql) [reply src/app/services/abuse-ip.service.ts]: There is a way by creating an interceptor: https://stackoverflow.com/questions/45735655/how-do-i-set-the-baseurl-for-angular-httpclient

**Review by diego-aquino — APPROVED** [review]

**Review by diego-aquino — APPROVED** [review]

**Pull request merged**